src/helper/HelperApp.cpp

```diff
@@ -193,8 +199,17 @@ namespace SDDM {
         return;
     }
 
-    void HelperApp::sessionFinished(int status) {
-        exit(status);
+    void HelperApp::sessionFinished(int exitCode, QProcess::ExitStatus exitStatus) {
+        if (exitStatus == QProcess::CrashExit) {
+            qWarning("Session crashed (killed by signal %d).", exitCode);
+            exit(Auth::HELPER_SESSION_ERROR);
+        }
+        else if (exitCode != 0) {
```

Comment:
Shouldn't this check for `NormalExit`?

Reply:
The first `if` branch already checks for `CrashExit` instead, if we run into this branch then we should be in the case of a `NormalExit` (as there are only two enum entries in `QProcess::ExitStatus`).
I doubt this enum would change in QtBase for future releases.

Reply:
I'll rearrange a bit to make it clearer then